algorithm/cgal/cgal.cc

```diff
@@ -3930,6 +3929,169 @@ void print_polygon_with_holes(
   std::cout << " }" << std::endl;
 }
 
+template <typename P>
+bool emitPolygonsWithHoles(const Plane& plane, const std::vector<P>& polygons,
+                           emscripten::val& emit_polygon,
+                           emscripten::val& emit_point) {
+  bool emitted = false;
+  for (const P& polygon : polygons) {
+    const auto& outer = polygon.outer_boundary();
+    emit_polygon(false);
+    emitted = true;
+    for (auto edge = outer.edges_begin(); edge != outer.edges_end(); ++edge) {
+      if (edge->source() == edge->target()) {
+        // Skip zero length edges.
```

Comment:
What does it mean for an edge to have zero length? Wouldn't that by definition not be contributing any new information so it could be skipped? 

...Or is that what's happening here

Reply:
Yeah, they shouldn't exist, really.